### PR TITLE
Expose Nvidia metrics as ConfigMap to allow enabling and disabling of individual metric

### PR DESCRIPTION
**Pull request by farhan5900**
**What type of PR is this?**

Bug

**What this PR does/ why we need it**:

Introduces a ConfigMap representing config file for the `nvidia-dcgm-exporter` to allow individual metrics to be enabled or disabled.

**Which issue(s) this PR fixes**:

[D2IQ-76955](https://jira.d2iq.com/browse/D2IQ-76955)

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

* [ ] *If a chart is changed, the chart version is correctly incremented.*
* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: staging/nvidia/charts/nvidia-dcgm-exporter/templates/default-counters.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    # DCGM_FI_DEV_NVLINK_RECOVERY_ERROR_COUNT_TOTAL, counter, Total number of NVLink recovery errors.
+    DCGM_FI_DEV_NVLINK_BANDWIDTH_TOTAL,            counter, Total number of NVLink bandwidth counters for all lanes
+
+    # VGPU License status,,
+    DCGM_FI_DEV_VGPU_LICENSE_STATUS, gauge, vGPU License status
+
+    # Remapped rows,,
+    DCGM_FI_DEV_UNCORRECTABLE_REMAPPED_ROWS, counter, Number of remapped rows for uncorrectable errors
+    DCGM_FI_DEV_CORRECTABLE_REMAPPED_ROWS,   counter, Number of remapped rows for correctable errors
+    DCGM_FI_DEV_ROW_REMAP_FAILURE,           gauge,   Whether remapping of rows has failed
```
Comment (tillt) [inline staging/nvidia/charts/nvidia-dcgm-exporter/templates/default-counters.yaml:72]: Where did we get those? Is that an official source that we can reference ie in a comment?

**Review by tillt — APPROVED** [review]
Awesome, thanks a lot for your quick fix @farhan5900.

File: staging/nvidia/charts/nvidia-dcgm-exporter/templates/default-counters.yaml (hunk 1 above)
Comment (farhan5900) [reply staging/nvidia/charts/nvidia-dcgm-exporter/templates/default-counters.yaml:72]: We can reference it from here: https://github.com/NVIDIA/gpu-monitoring-tools/blob/master/etc/dcgm-exporter/default-counters.csv. Right now, it is taken from the current `dcgm-exporter` deployed image.

**Review by mhrabovcin — APPROVED** [review]

File: staging/nvidia/charts/nvidia-dcgm-exporter/templates/default-counters.yaml (hunk 1 above)
Comment (alejandroEsc) [reply staging/nvidia/charts/nvidia-dcgm-exporter/templates/default-counters.yaml:72]: Can we please add a comment that references the above information?

File: staging/nvidia/charts/nvidia-dcgm-exporter/templates/default-counters.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: nvidia-dcgm-exporter-counters
+  namespace: {{ .Release.Namespace }}
+data:
+  default-counters.csv: |
```
Comment (joejulian) [inline staging/nvidia/charts/nvidia-dcgm-exporter/templates/default-counters.yaml:7]:
```suggestion
  default-counters.csv: |
    # These settings are taken from the dcgm-exporter  image
    # https://github.com/NVIDIA/gpu-monitoring-tools/blob/master/etc/dcgm-exporter/default-counters.csv
    #
```

File: staging/nvidia/charts/nvidia-dcgm-exporter/templates/default-counters.yaml (hunk 1 above)
Comment (farhan5900) [reply staging/nvidia/charts/nvidia-dcgm-exporter/templates/default-counters.yaml:75]: Added It.

File: staging/nvidia/charts/nvidia-dcgm-exporter/templates/default-counters.yaml (hunk 2 above)
Comment (alejandroEsc) [reply staging/nvidia/charts/nvidia-dcgm-exporter/templates/default-counters.yaml:7]: thank you @joejulian he did add it so marking this resolved if thats ok.

**Review by alejandroEsc — APPROVED** [review]
lgtm, thank you

**Pull request merged**